lib/ansible/modules/web_infrastructure/sentry_project.py

```diff
@@ -0,0 +1,155 @@
+    required: true
+
+  state:
+    description:
+      - The state of project.
+    required: true
+    choices: [ "present", "absent" ]
+
+  team:
+    description:
+      - A team you are in and which has an access for project manipulating.
+
+  url:
+    description:
+      - Sentry URL with scheme.
+    required: true
+
+author: "Mikhail Naletov (@okgolove)"
+'''
+
+EXAMPLES = '''
+# Create project "backend" in "example" organization and "senior" team
+- sentry_project:
+    api_token: 1234567890abcdwxyz
+    organization: example
+    project_name: backend
+    project_slug: backend
+    state: present
+    team: senior
+    url: https://sentry.example.com
+'''
+
+RETURN = ''' # '''
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_native
+
+
+def main():
+    arg_spec = dict(
+        api_token=dict(required=True),
+        organization=dict(required=True),
+        project_name=dict(required=True),
+        project_slug=dict(required=True),
+        state=dict(required=True, choices=['present', 'absent']),
+        team=dict(required=True),
+        url=dict(required=True)
+    )
+
+    module = AnsibleModule(argument_spec=arg_spec, supports_check_mode=False)
+
+    api_token = module.params['api_token']
+    organization = module.params['organization']
+    project_name = module.params['project_name']
+    project_slug = module.params['project_slug']
+    state = module.params['state']
+    team = module.params['team']
+    url = module.params['url']
+
+    def is_project_exists(project_name, organization, team, api_token):
+        response, info = fetch_url(
+            module, "{url}/api/0/teams/{organization}/{team}/projects/".format(
+                url=url, organization=organization, team=team),
+            headers={"Authorization": "Bearer {0}".format(api_token)},
+            method="GET")
+        if info["status"] != 200:
+            module.fail_json(msg=info["msg"])
+        body = json.loads(response.read())
+        return bool([x for x in body if x["name"] == project_name])
+
+    if state == 'present':
+        if is_project_exists(project_name, organization, team, api_token):
+            module.exit_json(changed=False)
+        else:
+            response, info = fetch_url(
+                module, "{url}/api/0/teams/{organization}/{team}/projects/".format(
+                    url=url, organization=organization, team=team
+                ),
+                headers={
+                    "Authorization": "Bearer {0}".format(api_token),
+                    "Content-Type": "application/json"
+                },
+                data=json.dumps({
+                    "name": project_name,
+                    "slug": project_slug
+                }),
+                method="POST")
+            if info["status"] != 201:
+                module.fail_json(msg=info["msg"])
```

Comment:
Might want to add some sanity in case "msg" is not filled (i.e. during a 404, or 503)

Reply:
I don't think this one has been addressed yet.

Reply:
I've added some additional information.
https://github.com/ansible/ansible/pull/52944/files#diff-6e038abd617da7a554dbdfcf20b9b1a0R149

Isn't it enough?

Reply:
The probem is not the additional information, but that `"msg"` will not necessarily be a key in `info`, and thus `info["msg"]` will result in a `KeyError` raised. Better use `info.get('msg', '—')` instead (or something like that, in this case `—` will be shown if `msg` isn't available).

Reply:
I got your idea.

Reply:
I think I've resolved the thing we're talking about.